OgreMain/include/OgreSceneManager.h

```diff
@@ -944,6 +948,8 @@ namespace Ogre {
 
         CompositorChain* mActiveCompositorChain;
         bool mLateMaterialResolving;
+        typedef std::deque<String> ActiveMaterialSchemeStack;
+        ActiveMaterialSchemeStack mActiveMaterialSchemeStack;
```

Comment:
why do you introduce this instead of using the actual function-call stack?

Reply:
Mainly because `Root::_pushCurrentSceneManager(SceneManager* sm)` uses same logic and I didn't know if it actually is possible end up there from somewhere else and mess it up. `SceneManager::_render()` loop is extremely complicated when combined with the possibility of another scenemanager being called from the main scene manager and them having their own render calls which all end up in `SceneManager::_render()` as well.